src/components/Notifications/NotificationsList.tsx

```diff
@@ -342,43 +329,73 @@ export default function NotificationsList({
 
   async function subscribe() {
     setIsSubscribing(true);
-    const response = await request(routes.getPublicKey);
-    const public_key = response.data?.public_key;
-    const sw = await navigator.serviceWorker.ready;
-    const push = await sw.pushManager.subscribe({
-      userVisibleOnly: true,
-      applicationServerKey: public_key,
-    });
-    const p256dh = btoa(
-      String.fromCharCode.apply(
-        null,
-        new Uint8Array(push.getKey("p256dh") as any) as any,
-      ),
-    );
-    const auth = btoa(
-      String.fromCharCode.apply(
-        null,
-        new Uint8Array(push.getKey("auth") as any) as any,
-      ),
-    );
+    try {
+      const response = await request(routes.getPublicKey);
+      const public_key = response.data?.public_key;
+      const sw = await navigator.serviceWorker.ready;
+      const push = await sw.pushManager.subscribe({
+        userVisibleOnly: true,
+        applicationServerKey: public_key,
+      });
+      const p256dh = btoa(
+        String.fromCharCode.apply(
+          null,
+          new Uint8Array(push.getKey("p256dh") as any) as any,
+        ),
+      );
+      const auth = btoa(
+        String.fromCharCode.apply(
+          null,
+          new Uint8Array(push.getKey("auth") as any) as any,
+        ),
+      );
 
-    const data = {
-      pf_endpoint: push.endpoint,
-      pf_p256dh: p256dh,
-      pf_auth: auth,
-    };
+      const data = {
+        pf_endpoint: push.endpoint,
+        pf_p256dh: p256dh,
+        pf_auth: auth,
+      };
 
-    const { res } = await request(routes.updateUserPnconfig, {
-      pathParams: { username: username },
-      body: data,
-    });
+      const { res } = await request(routes.updateUserPnconfig, {
+        pathParams: { username: username },
+        body: data,
+      });
+
+      if (res?.ok) {
+        setIsSubscribed("SubscribedOnThisDevice");
+        Success({
+          msg: t("subscribed_successfully"),
+        });
+        setIsSubscribing(false);
+      } else {
+        Error({
+          msg: t("subscription_failed"),
+        });
+        setIsSubscribing(false);
+      }
+    } catch (error) {
+      const permission = Notification.permission;
 
-    if (res?.ok) {
-      setIsSubscribed("SubscribedOnThisDevice");
+      if (permission === "denied") {
+        Warn({
+          msg: t("notification_permission_denied"),
+        });
+        setIsSubscribing(false);
+        return;
+      } else if (permission === "default") {
+        Warn({
+          msg: t("notification_cancelled"),
```

Comment:
Let this give the permission denied error only.

Reply:
Ok